Formula/ddd.rb

```diff
@@ -31,34 +30,8 @@ class Ddd < Formula
   depends_on "libxt"
   depends_on "openmotif"
 
-  # https://savannah.gnu.org/bugs/?41997
-  patch do
-    url "https://savannah.gnu.org/patch/download.php?file_id=31132"
-    sha256 "f3683f23c4b4ff89ba701660031d4b5ef27594076f6ef68814903ff3141f6714"
-  end
-
-  # Patch to fix compilation with Xcode 9
-  # https://savannah.gnu.org/bugs/?52175
-  patch :p0 do
-    url "https://raw.githubusercontent.com/macports/macports-ports/a71fa9f4/devel/ddd/files/patch-unknown-type-name-a_class.diff"
-    sha256 "c187a024825144f186f0cf9cd175f3e972bb84590e62079793d0182cb15ca183"
-  end
-
-  # Patch to fix compilation with Xt 1.2.0
-  # https://savannah.gnu.org/patch/?9992
-  patch do
-    url "https://savannah.gnu.org/patch/download.php?file_id=50229"
-    sha256 "140a1493ab640710738abf67838e63fbb8328590d1d3ab0212e7ca1f378a9ee7"
-  end
-
   def install
-    if OS.linux?
-      # Patch to fix compilation error
-      # https://savannah.gnu.org/bugs/?33960
-      # Remove with next release
-      inreplace "ddd/strclass.C", "#include <stdlib.h>", "#include <stdlib.h>\n#include <cstdio>"
-    end
-
+    ENV.append_to_cflags "-DHAVE_SYS_IOCTL_H" if OS.mac?
```

Comment:
If this works, this needs to be reported upstream. The macro in `ddd/agent/TTYAgent.C` checks for `HAVE_SYS_IOCTL_H`, but given their `configure` script it probably needs to check for `HAVE_IOCTL_DECL` or something. (Not quite sure since I don't follow their autoconf macros 100%.)

Reply:
Looks like it worked. I'll open an issue upstream.